File: Models/Monica_SoilTemp/src/cpp/Monica_SoilTemp/SoilTemperatureCompExogenous.cpp
#include "SoilTemperatureCompExogenous.h"
using namespace Monica_SoilTemp;


SoilTemperatureCompExogenous::SoilTemperatureCompExogenous() {}

double SoilTemperatureCompExogenous::gettmin() { return this->tmin; }
double SoilTemperatureCompExogenous::gettmax() { return this->tmax; }
double SoilTemperatureCompExogenous::getglobrad() { return this->globrad; }
double SoilTemperatureCompExogenous::getsoilCoverage() { return this->soilCoverage; }
double SoilTemperatureCompExogenous::getsoilSurfaceTemperatureBelowSnow() { return this->soilSurfaceTemperatureBelowSnow; }
bool SoilTemperatureCompExogenous::gethasSnowCover() { return this->hasSnowCover; }

void SoilTemperatureCompExogenous::settmin(double _tmin) { this->tmin = _tmin; }
void SoilTemperatureCompExogenous::settmax(double _tmax) { this->tmax = _tmax; }
void SoilTemperatureCompExogenous::setglobrad(double _globrad) { this->globrad = _globrad; }
void SoilTemperatureCompExogenous::setsoilCoverage(double _soilCoverage) { this->soilCoverage = _soilCoverage; }
void SoilTemperatureCompExogenous::setsoilSurfaceTemperatureBelowSnow(double _soilSurfaceTemperatureBelowSnow) { this->soilSurfaceTemperatureBelowSnow = _soilSurfaceTemperatureBelowSnow; }
void SoilTemperatureCompExogenous::sethasSnowCover(bool _hasSnowCover) { this->hasSnowCover = _hasSnowCover; }

File: Models/Monica_SoilTemp/src/java/Monica_SoilTemp/SoilTemperatureCompExogenous.java
import  java.io.*;
import  java.util.*;
import java.time.LocalDateTime;

public class SoilTemperatureCompExogenous
{
    private Double tmin;
    private Double tmax;
    private Double globrad;
    private Double soilCoverage;
    private Double soilSurfaceTemperatureBelowSnow;
    private Boolean hasSnowCover;
    
    public SoilTemperatureCompExogenous() { }
    
    public SoilTemperatureCompExogenous(SoilTemperatureCompExogenous toCopy, boolean copyAll) // copy constructor 
    {
        if (copyAll)
        {
            this.tmin = toCopy.gettmin();
            this.tmax = toCopy.gettmax();
            this.globrad = toCopy.getglobrad();
            this.soilCoverage = toCopy.getsoilCoverage();
            this.soilSurfaceTemperatureBelowSnow = toCopy.getsoilSurfaceTemperatureBelowSnow();
            this.hasSnowCover = toCopy.gethasSnowCover();
        }
    }
    public Double gettmin()
    { return tmin; }

    public void settmin(Double _tmin)
    { this.tmin= _tmin; } 
    
    public Double gettmax()
    { return tmax; }

    public void settmax(Double _tmax)
    { this.tmax= _tmax; } 
    
    public Double getglobrad()
    { return globrad; }

    public void setglobrad(Double _globrad)
    { this.globrad= _globrad; } 
    
    public Double getsoilCoverage()
    { return soilCoverage; }

    public void setsoilCoverage(Double _soilCoverage)
    { this.soilCoverage= _soilCoverage; } 
    
    public Double getsoilSurfaceTemperatureBelowSnow()
    { return soilSurfaceTemperatureBelowSnow; }

    public void setsoilSurfaceTemperatureBelowSnow(Double _soilSurfaceTemperatureBelowSnow)
    { this.soilSurfaceTemperatureBelowSnow= _soilSurfaceTemperatureBelowSnow; } 
    
    public Boolean gethasSnowCover()
    { return hasSnowCover; }

    public void sethasSnowCover(Boolean _hasSnowCover)
    { this.hasSnowCover= _hasSnowCover; } 
    
}